magento/magento2
# ISSUE-16315 - add prevent for ignoring product website assignments in…

… "Update on save" product flat indexer mode





### Description (*)

This fix loads only store views that are connected to product to avoid save reindexed row to all product flat tables in the database. The problem occurred only in "Update on save" mode for "Product Flat Data" indexer.

### Fixed Issues (if relevant)

1. magento/magento2#16315: Product save with onthefly index ignores website assignments

### Manual testing scenarios (*)

1. Create 2 websites and 2 store views
2. Assign first store view to first website and second store view to second website
3. Add product and assign only to first website
4. Save product
5. Check if product is assigned to second store view 

### Questions or comments


### Contribution checklist (*)
